### PR TITLE
Ref/cmd output

### PR DESCRIPTION
**Pull request by thomastaylor312**
**What this PR does / why we need it**:
This refactors the work done in #5593 by @deschmih into some more encapsulated functions to improve code reuse. It also restores the unexported `PrintStatus` function to be exported again

**Special notes for your reviewer**:
Please DO NOT merge this until #5593 is merged and I rebase on master

### CODE REVIEW

File: cmd/helm/install.go [hunk 1]
```diff
@@ -335,7 +337,10 @@ func (i *installCmd) run() error {
 	if rel == nil {
 		return nil
 	}
-	i.printRelease(rel)
+
+	if i.output == "table" {
```
Comment (bacongobbler) [inline cmd/helm/install.go]: can we check for equality with `string(outputTable)` here?

File: cmd/helm/upgrade.go [hunk 2]
```diff
@@ -307,14 +309,14 @@ func (u *upgradeCmd) run() error {
 		printRelease(u.out, resp.Release)
 	}
 
-	fmt.Fprintf(u.out, "Release %q has been upgraded.\n", u.release)
-
+	if u.output == "table" {
```
Comment (bacongobbler) [inline cmd/helm/upgrade.go]: same here RE: checking against `outputTable`

**Comment by bacongobbler** [comment]
I just realized those two comments are from code in #5593. I'll comment on that PR instead.

**Review by bacongobbler — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]
LGTM, nice tidy up @thomastaylor312. Good to merge.

**Comment by thomastaylor312** [comment]
Ported to Helm 3 with #6504

**Comment by parubets** [comment]
just updated to 2.15, having problems with helm-upgrade:
if i do upgrade with no prior installation but specify option --install i get this error:

`helm upgrade staging-rates ./.deploy/k8s --namespace staging -f ./.deploy/k8s/values/staging.yaml --install --wait
Release "staging-rates" does not exist. Installing it now.
Error: unsupported format
make: *** [helm-upgrade] Error 1`

used to work fine before this code change

**Comment by thomastaylor312** [comment]
@parubets I can't seem to duplicate the issue. Could it be something else you have saved in your makefile? Feel free to open a separate issue and we can converse more there

**Pull request merged**